configuration/device-tree.md

```diff
@@ -332,10 +310,85 @@ Note that a property is assigned the value `false` by not defining it. Boolean p
 name = <&label>,"property?";
 ```
 
-where `label` and `property` are replaced by suitable values. Boolean parameters can cause properties to be created or deleted.
+where `label` and `property` are replaced by suitable values.
+
+Inverted booleans invert the input value before applying it in the same was as a regular boolean; they are declared similarly, but use `!` to indicate the inversion:
+
+```
+name = <&label>,"property!";
+```
 
+Boolean parameters can cause properties to be created or deleted, but they can't delete a property that already exists in the base DTB.
+
+#### 2.2.4: Byte string parameters
 <a name="part2.2.4"></a>
-#### 2.2.4 Overlay/fragment parameters
+
+Byte string properties are arbitrary sequences of bytes, e.g. MAC addresses. They accept strings of hexadecimal bytes, with or without colons between the bytes.
+
+```
+mac_address = <&ethernet0>,"local_mac_address[";
+```
+The `[` was chosen to match the DT syntax for declaring a byte string:
+```
+local_mac_address = [aa bb cc dd ee ff];
+```
+
+#### 2.2.5: Parameters with multiple targets
+<a name="part2.2.5"></a>
+
+There are some situations where it is convenient to be able to set the same value in multiple locations within the Device Tree. Rather than the ungainly approach of creating multiple parameters, it is possible to add multiple targets to a single parameter by concatenating them, like this:
+
+```
+    __overrides__ {
+        gpiopin = <&w1>,"gpios:4",
+                  <&w1_pins>,"brcm,pins:0";
+        ...
+    };
+```
+
+(example taken from the `w1-gpio` overlay)
+
+Note that it is even possible to target properties of different types with a single parameter. You could reasonably connect an "enable" parameter to a `status` string, cells containing zero or one, and a proper boolean property.
+
+#### 2.2.6: Literal assignments
+<a name="part2.2.6"></a>
+
+As seen in <a href="part2.2.5">2.2.5</a>, the DT parameter mechanism allows multiple targets to be patched from the same parameter, but the utility is limited by the fact that the same value has to be written to all locations (except for format conversion and the negation available from inverted booleans). The addition of embedded literal assignments allows a parameter to write arbitrary values, regardless of the parameter value supplied by the user.
+
+Assignments appear at the end of a declaration, and are indicated by a `=`:
+
+```
+str_val  = <&target>,"strprop=value";              // 1
+int_val  = <&target>,"intprop:0=42                 // 2
+int_val2 = <&target>,"intprop:0=",<42>;            // 3
+bytes    = <&target>,"bytestr[=b8:27:eb:01:23:45"; // 4
+```
+Lines 1, 2 and 4 are fairly obvious, but line 3 is more interesting because the value appears as an integer (cell) value. The DT compiler evaluates integer expressions at compile time, which might be convenient (particularly if macro values are used), but the cell can also contain a reference to a label:
```

Comment:
Could you clarify if/how this makes the effects of lines 2 and 3 different? Or is this just a pre-cursor to the following paragraph?

Reply:
The end results of lines 2 and 3 are the same, but line 3 could have been written with `,<(6*9)>`, or `<MEANING_OF_LIFE>` with the appropriate macro constant definition. But yes, it is largely a set-up for the next paragraph on labels,